### PR TITLE
WIP: Espnet Multi-convformer implementation

### PR DESCRIPTION
**Pull request by Darshan7575**
## What?
* This PR adds a new encoder architecture coined Multi-convformer that extends conformer with [gMLP](https://arxiv.org/abs/2105.08050) style gating and a multi-convolution setup. More details about this architecture can be found [here](https://arxiv.org/abs/2407.03718).
* It also adds recipes and our results w/ links to models to Librispeech-100, Librispeech and SLURP datasets.

## Known issues:
* Turns out, grouped 2d convolutions are slower in PyTorch v1.x, and it has not yet been resolved to date (see [here](https://github.com/pytorch/pytorch/issues/73764)). Since some of our code relies on this, we recommend using PyTorch v2 when training on large datasets. 

## Citation:
If you use our work, please consider citing us:
```bibtex
@misc{prabhu2024multiconvformerextendingconformermultiple,
      title={Multi-Convformer: Extending Conformer with Multiple Convolution Kernels}, 
      author={Darshan Prabhu and Yifan Peng and Preethi Jyothi and Shinji Watanabe},
      year={2024},
      eprint={2407.03718},
      archivePrefix={arXiv},
      primaryClass={cs.CL},
      url={https://arxiv.org/abs/2407.03718}, 
}
```

### CODE REVIEW

**Comment by sw005320** [comment]
@pyf98, can you review this PR?

**Comment by sw005320** [comment]
@pyf98, this is a reminder.

**Comment by sw005320** [comment]
Please check https://github.com/espnet/espnet/actions/runs/9908186288/job/27373422523#step:8:292

File: espnet2/asr/encoder/multiconvformer_encoder.py [hunk 1]
```diff
@@ -0,0 +1,404 @@
+# Copyright 2020 Tomoki Hayashi
```
Comment (pyf98) [inline espnet2/asr/encoder/multiconvformer_encoder.py:1]: Is this line correct?

**Comment by pyf98** [comment]
Looks good to me.

**Comment by sw005320** [comment]
@Darshan7575, I observed that the test coverage by `codecov` was dropped.
It might be an issue in `codecov` since you added the unit test, but can you double-check it?
Then, I'll merge this PR.

**Comment by Darshan7575** [comment]
I couldn't find any apparent problems with the new unit test I added. Plus, the unit test I wrote should cover almost all aspects of the newly defined classes. Weirdly, the `codecov/patch` test coverage fails even before all the unit tests finish. It seems like a problem with `codecov`.

**Comment by sw005320** [comment]
Thanks!
Then, I'll merge it.

**Pull request merged**